src/nccl_ofi_net.c

```diff
@@ -1443,11 +1489,13 @@ static ncclResult_t ofi_connect(int dev, void *handle, void **sendComm)
 	} while (true);
 
 	/* Ensure the message is sent. */
-	do {
-		ret = nccl_ofi_progress(nccl_ofi_component[dev]);
-		if (OFI_UNLIKELY(ret != 0))
+	/* Skip this step for sending to self */
+	if (!conn_info->connect_to_self) {
+		ret = finalize_connection_info(conn_info);
```

Comment:
rename this function to `process_connect_completion`?

Reply:
done